File: FalconMindBuilder/frontend/src/api/projects.ts
/**
 * Projects API
 */
import apiClient from './client'

export interface Project {
  id: string
  name: string
  description?: string
  uav_id?: string
  created_at: string
  updated_at: string
  flows_count: number
}

export interface ProjectCreate {
  name: string
  description?: string
  uav_id?: string
}

export interface ProjectUpdate {
  name?: string
  description?: string
  uav_id?: string
}

export const projectsApi = {
  /**
   * List all projects
   */
  async list(): Promise<Project[]> {
    const response = await apiClient.get('/projects/')
    return response
  },

  /**
   * Create a new project
   */
  async create(data: ProjectCreate): Promise<Project> {
    const response = await apiClient.post('/projects/', data)
    return response
  },

  /**
   * Get project by ID
   */
  async get(projectId: string): Promise<Project> {
    const response = await apiClient.get(`/projects/${projectId}`)
    return response
  },

  /**
   * Update project
   */
  async update(projectId: string, data: ProjectUpdate): Promise<Project> {
    const response = await apiClient.put(`/projects/${projectId}`, data)
    return response
  },

  /**
   * Delete project
   */
  async delete(projectId: string): Promise<void> {
    await apiClient.delete(`/projects/${projectId}`)
  }
}


// 独立导出函数（用于 TemplateWizard）
export const getProjects = projectsApi.list
export const getProject = projectsApi.get
export const createProject = projectsApi.create
export const updateProject = projectsApi.update
export const deleteProject = projectsApi.delete

File: FalconMindBuilder/backend/app/api/projects.py
"""
Project API Routes
"""
from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.orm import Session
from typing import List

from ..core.database import get_db
from ..models.project import Project
from ..schemas.project import ProjectCreate, ProjectUpdate, ProjectResponse

router = APIRouter(prefix="/api/projects", tags=["projects"])


@router.get("/", response_model=List[ProjectResponse])
async def list_projects(db: Session = Depends(get_db)):
    """List all projects"""
    projects = db.query(Project).order_by(Project.created_at.desc()).all()
    return [p.to_dict() for p in projects]


@router.post("/", response_model=ProjectResponse, status_code=status.HTTP_201_CREATED)
async def create_project(
    project: ProjectCreate,
    db: Session = Depends(get_db)
):
    """Create a new project"""
    db_project = Project(
        name=project.name,
        description=project.description,
        uav_id=project.uav_id
    )
    
    db.add(db_project)
    db.commit()
    db.refresh(db_project)
    
    return db_project.to_dict()


@router.get("/{project_id}", response_model=ProjectResponse)
async def get_project(project_id: str, db: Session = Depends(get_db)):
    """Get project by ID"""
    project = db.query(Project).filter(Project.id == project_id).first()
    if not project:
        raise HTTPException(status_code=404, detail="Project not found")
    return project.to_dict()


@router.put("/{project_id}", response_model=ProjectResponse)
async def update_project(
    project_id: str,
    project_update: ProjectUpdate,
    db: Session = Depends(get_db)
):
    """Update project"""
    project = db.query(Project).filter(Project.id == project_id).first()
    if not project:
        raise HTTPException(status_code=404, detail="Project not found")
    
    # Update fields
    update_data = project_update.model_dump(exclude_unset=True)
    for field, value in update_data.items():
        setattr(project, field, value)
    
    db.commit()
    db.refresh(project)
    
    return project.to_dict()


@router.delete("/{project_id}", status_code=status.HTTP_204_NO_CONTENT)
async def delete_project(project_id: str, db: Session = Depends(get_db)):
    """Delete project"""
    project = db.query(Project).filter(Project.id == project_id).first()
    if not project:
        raise HTTPException(status_code=404, detail="Project not found")
    
    db.delete(project)
    db.commit()
    
    return None
